File: src/main/webapp/app/entities/reports/services/sales-forecast.service.ts
import { Injectable, inject } from '@angular/core';
import { HttpClient, HttpResponse, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { SERVER_API_URL } from 'app/app.constants';
import { ISalesForecast, IForecastSummary } from 'app/shared/model/report/sales-forecast.model';

type ForecastResponseType = HttpResponse<ISalesForecast[]>;
type SummaryResponseType = HttpResponse<IForecastSummary>;

/**
 * Service for Sales Forecasting
 */
@Injectable({ providedIn: 'root' })
export class SalesForecastService {
  private readonly resourceUrl = SERVER_API_URL + 'api/sales-forecast';
  private readonly http = inject(HttpClient);

  /**
   * Get sales forecast
   */
  getForecast(monthsAhead: number, method: string): Observable<ForecastResponseType> {
    const params = new HttpParams().set('monthsAhead', monthsAhead.toString()).set('method', method);
    return this.http.get<ISalesForecast[]>(this.resourceUrl, { params, observe: 'response' });
  }

  /**
   * Get forecast summary
   */
  getSummary(): Observable<SummaryResponseType> {
    return this.http.get<IForecastSummary>(`${this.resourceUrl}/summary`, { observe: 'response' });
  }

  /**
   * Get historical vs forecast comparison
   */
  getHistoricalVsForecast(startDate: string, endDate: string, monthsAhead: number): Observable<ForecastResponseType> {
    const params = new HttpParams()
      .set('startDate', startDate)
      .set('endDate', endDate)
      .set('monthsAhead', monthsAhead.toString());
    return this.http.get<ISalesForecast[]>(`${this.resourceUrl}/historical`, { params, observe: 'response' });
  }

  /**
   * Detect seasonality
   */
  detectSeasonality(): Observable<HttpResponse<boolean>> {
    return this.http.get<boolean>(`${this.resourceUrl}/seasonality`, { observe: 'response' });
  }
}

File: src/main/java/com/kobe/warehouse/service/report/SalesForecastService.java
package com.kobe.warehouse.service.report;

import com.kobe.warehouse.service.dto.report.ForecastSummaryDTO;
import com.kobe.warehouse.service.dto.report.SalesForecastDTO;
import java.time.LocalDate;
import java.util.List;

/**
 * Service interface for Sales Forecasting
 */
public interface SalesForecastService {

    /**
     * Get sales forecast for next N months
     * @param monthsAhead number of months to forecast (3, 6, or 12)
     * @param method forecasting method (LINEAR_REGRESSION, MOVING_AVERAGE, SEASONAL)
     * @return list of forecast data points
     */
    List<SalesForecastDTO> getForecast(Integer monthsAhead, String method);

    /**
     * Get forecast summary with statistics
     * @return summary with totals, trends, and accuracy metrics
     */
    ForecastSummaryDTO getForecastSummary();

    /**
     * Get historical data vs forecast for comparison
     * @param startDate start date for historical data
     * @param endDate end date for historical data
     * @return list combining historical and forecast data
     */
    List<SalesForecastDTO> getHistoricalVsForecast(LocalDate startDate, LocalDate endDate, Integer monthsAhead);

    /**
     * Detect seasonality in historical sales data
     * @return true if seasonality detected
     */
    Boolean detectSeasonality();
}
